src/session/session_api.c

```diff
@@ -527,10 +531,18 @@ static int
 __session_create_readonly(
     WT_SESSION *wt_session, const char *uri, const char *config)
 {
+	WT_DECL_RET;
+	WT_SESSION_IMPL *session;
+
 	WT_UNUSED(uri);
 	WT_UNUSED(config);
 
-	return (__wt_session_notsup(wt_session));
+	session = (WT_SESSION_IMPL *)wt_session;
+	SESSION_API_CALL_NOCONF(session, create);
+
+	WT_STAT_FAST_CONN_INCR(session, session_table_create_fail);
+	ret = __wt_session_notsup(wt_session);
```

Comment:
All callers of ``__wt_session_notsup`` now have a ``WT_SESSION_IMPL`` - it would be better to change the parameter type to the function rather than pass in a ``WT_SESSION``